libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/SsmConfigService.java

```diff
@@ -0,0 +1,151 @@
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient;
+import software.amazon.awssdk.services.secretsmanager.model.DecryptionFailureException;
+import software.amazon.awssdk.services.secretsmanager.model.InternalServiceErrorException;
+import software.amazon.awssdk.services.secretsmanager.model.InvalidParameterException;
+import software.amazon.awssdk.services.secretsmanager.model.InvalidRequestException;
+import software.amazon.awssdk.services.secretsmanager.model.ResourceNotFoundException;
+import software.amazon.awssdk.services.ssm.SsmClient;
+import software.amazon.lambda.powertools.parameters.ParamManager;
+import software.amazon.lambda.powertools.parameters.SSMProvider;
+import software.amazon.lambda.powertools.parameters.SecretsProvider;
+import uk.gov.di.ipv.core.library.annotations.ExcludeFromGeneratedCoverageReport;
+import uk.gov.di.ipv.core.library.helpers.LogHelper;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+
+import static java.time.temporal.ChronoUnit.MINUTES;
+import static uk.gov.di.ipv.core.library.config.EnvironmentVariable.CONFIG_SERVICE_CACHE_DURATION_MINUTES;
+import static uk.gov.di.ipv.core.library.config.EnvironmentVariable.ENVIRONMENT;
+import static uk.gov.di.ipv.core.library.helpers.LogHelper.LogField.LOG_FEATURE_SET;
+import static uk.gov.di.ipv.core.library.helpers.LogHelper.LogField.LOG_MESSAGE_DESCRIPTION;
+import static uk.gov.di.ipv.core.library.helpers.LogHelper.LogField.LOG_PARAMETER_PATH;
+import static uk.gov.di.ipv.core.library.helpers.LogHelper.LogField.LOG_SECRET_ID;
+
+public class SsmConfigService extends ConfigService {
+
+    private static final String CORE_BASE_PATH = "/%s/core/";
+    private static final int DEFAULT_CACHE_DURATION_MINUTES = 3;
+    private static final Logger LOGGER = LogManager.getLogger();
+    private final SSMProvider ssmProvider;
+    private final SecretsProvider secretsProvider;
+
+    public SsmConfigService(
+            SSMProvider ssmProvider, SecretsProvider secretsProvider, List<String> featureSet) {
+        this.ssmProvider = ssmProvider;
+        this.secretsProvider = secretsProvider;
+        setFeatureSet(featureSet);
+    }
+
+    public SsmConfigService(SSMProvider ssmProvider, SecretsProvider secretsProvider) {
+        this(ssmProvider, secretsProvider, null);
+    }
+
+    @ExcludeFromGeneratedCoverageReport
+    public SsmConfigService() {
+        var cacheDuration =
+                getEnvironmentVariable(CONFIG_SERVICE_CACHE_DURATION_MINUTES) == null
+                        ? DEFAULT_CACHE_DURATION_MINUTES
+                        : Integer.parseInt(
+                                getEnvironmentVariable(CONFIG_SERVICE_CACHE_DURATION_MINUTES));
+
+        this.ssmProvider =
+                ParamManager.getSsmProvider(
+                                SsmClient.builder()
+                                        .httpClient(UrlConnectionHttpClient.create())
```

Comment:
Did we ever put back the crt client? In fact - did we ever get anywhere with the segfaults?

Reply:
I tried the async CRT client and it did the same thing :(

We have a ticket to try again once the dynatrace layer has been updated, https://govukverify.atlassian.net/browse/PYIC-6958 - DT seems to be a contributing factor so hoping this might just fix it... (I'm not holding my breath)